servatrice/src/serversocketinterface.cpp

```diff
@@ -877,63 +877,77 @@ Response::ResponseCode AbstractServerSocketInterface::cmdRegisterAccount(const C
     QString clientId = QString::fromStdString(cmd.clientid());
     qDebug() << "Got register command: " << userName;
 
+    Response_Register *re = new Response_Register;
+    re->set_custom_reason_str(servatrice->getCustomRegErrorMessage().toStdString());
+
     bool registrationEnabled = settingsCache->value("registration/enabled", false).toBool();
-    if (!registrationEnabled)
+    if (!registrationEnabled) {
+        rc.setResponseExtension(re);
         return Response::RespRegistrationDisabled;
+    }
 
     QString emailAddress = QString::fromStdString(cmd.email());
     bool requireEmailForRegistration = settingsCache->value("registration/requireemail", true).toBool();
     if (requireEmailForRegistration)
     {
         QRegExp rx("\\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,4}\\b");
-        if(emailAddress.isEmpty() || !rx.exactMatch(emailAddress))
+        if (emailAddress.isEmpty() || !rx.exactMatch(emailAddress)) {
+            rc.setResponseExtension(re);
             return Response::RespEmailRequiredToRegister;
+        }
     }
 
     // TODO: Move this method outside of the db interface
     QString errorString;
     if (!sqlInterface->usernameIsValid(userName, errorString))
     {
-        Response_Register *re = new Response_Register;
         re->set_denied_reason_str(errorString.toStdString());
         rc.setResponseExtension(re);
         return Response::RespUsernameInvalid;
     }
 
-    if (userName.toLower().simplified() == "servatrice")
+    if (userName.toLower().simplified() == "servatrice") {
```

Comment:
We actually use this username to send server messages and is why the name is hard denied for registration.